packages/fluentui/react-bindings/src/hooks/useStyles.ts

```diff
@@ -63,22 +65,34 @@ const useStyles = <StyleProps extends PrimitiveProps>(
   const context: StylesContextValue<{ renderRule: RendererRenderRule }> =
     React.useContext(ThemeContext) || defaultContext;
 
+  const [, componentProps] = useReactElement();
+  const composeOptions = useComposeOptions();
+
   const {
     className = process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉',
     mapPropsToStyles = () => ({} as StyleProps),
     mapPropsToInlineStyles = () => ({} as InlineStyleProps<StyleProps>),
     rtl = false,
   } = options;
+  const componentStylesProps = mapPropsToStyles();
+
+  // `composeProps` should include all props including stylesProps as they can contain state
+  const composeProps = { ...componentProps, ...componentStylesProps };
```

Comment:
Just to clarify, we have here all styles props from the prev component + this component's props. We do not have the state from the component, it was not assigned previously to some styles props?

Reply:
> We do not have the state from the component, it was not assigned previously to some styles props?

Yep, only styles props that in host components were mapped from state

Reply:
I hope this won't be a problem :\